handlers/handler-bizkeeper/src/main/java/org/apache/servicecomb/bizkeeper/event/CircutBreakerEventNotifier.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.bizkeeper.event;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.servicecomb.foundation.common.event.EventManager;
+import org.apache.servicecomb.foundation.common.event.AlarmEvent.Type;
+
+import com.netflix.hystrix.HystrixCommandKey;
+import com.netflix.hystrix.HystrixEventType;
+import com.netflix.hystrix.strategy.eventnotifier.HystrixEventNotifier;
+
+public class CircutBreakerEventNotifier extends HystrixEventNotifier {
+
+  /**
+   * 使用circuitMarker来记录被熔断的接口
+   */
+  private static ConcurrentHashMap<String, Boolean> circuitMarker = new ConcurrentHashMap<>();
+
+  @Override
+  public void markEvent(HystrixEventType eventType, HystrixCommandKey key) {
+    String keyName = key.name();
+    switch (eventType) {
+      case SHORT_CIRCUITED:
+        if (circuitMarker.get(keyName) == null) {
```

Comment:
concurrency problem
i had wrote this problem days ago.

Reply:
remember you said that, subscribers not only need switch event, but also every invocation that circuited.

Reply:
subscribers only need switch event， We need only send event when last time is OPEN and turn to be CLOSED. Can't send for each invocation.I discussed with Liu Bao.
about concurrency problem，can you provide some improvement suggestions?

Reply:
ConcurrentHashMapEx<String, AtomicBoolean> circuitFlag = new ConcurrentHashMapEx<>();
    AtomicBoolean flag = circuitFlag.computeIfAbsent(key, k -> new AtomicBoolean());
    // circuit event
    if (flag.compareAndSet(false, true)) {
      // post xxx event
    }

    // success event
    if (flag.compareAndSet(true, false)) {
      // post xxx event
    }
-----------
and infact, i don't know what does eventType mean, open/close/succeeded/failed

Reply:
Succeeded/failed is not used and has been deleted. Open/close is used to indicate circuit and isolation status.